### PR TITLE
Add support for parameters in types

### PR DESCRIPTION
**Pull request by Mingun**
Even `ksv` has no such feature (see https://github.com/kaitai-io/kaitai_struct_visualizer/issues/33). Now you can look at parameterized types somehow:

parameters

Based on #6, so can be merged instead of #6.

### CODE REVIEW

**Comment by GreyCat** [comment]
Can I ask you to rebase it against latest master and we'll merge it right away?

**Comment by Mingun** [comment]
Rebased and CI is also fixed

File: src/main/java/io/kaitai/struct/visualizer/VisualizerPanel.java [hunk 1]
```diff
@@ -34,7 +39,17 @@
...
-    private static final Pattern TOP_CLASS_NAME = Pattern.compile("public class (.*?) extends KaitaiStruct");
+    /**
+     * Regexp with 2 groups: class name and type parameters. Type parameters
+     * must be parsed with {@link #PARAMETER_NAME}.
+     */
+    private static final Pattern TOP_CLASS_NAME_AND_PARAMETERS = Pattern.compile(
+        "public class (.+?) extends KaitaiStruct.*" +
+        "public \\1\\(KaitaiStream _io, KaitaiStruct _parent, \\1 _root(.*?)\\)",
+        Pattern.DOTALL
+    );
```
Comment (GreyCat) [inline src/main/java/io/kaitai/struct/visualizer/VisualizerPanel.java]: Actually, nowadays top class name should not be that hard to get. Probably we should expose [JavaMain.compileOneInput](https://github.com/kaitai-io/kaitai_struct_compiler/blob/master/jvm/src/main/scala/io/kaitai/struct/JavaMain.scala#L277) that will clearly return compilation result as Java object, and that object [contains reference to top class name](https://github.com/kaitai-io/kaitai_struct_compiler/blob/master/jvm/src/main/scala/io/kaitai/struct/JavaMain.scala#L291).

**Comment by GreyCat** [comment]
Thanks for this contribution!

**Pull request merged**